src/managers/AutoModerationRuleManager.js

```diff
@@ -0,0 +1,289 @@
+   */
+
+  /**
+   * Options used to set the metadata of an auto moderation rule action.
+   * @typedef {Object} AutoModerationActionMetadataOptions
+   * @property {GuildTextChannelResolvable|ThreadChannel} [channel] The channel to which content will be logged
+   * @property {number} [durationSeconds] The timeout duration in seconds
+   */
+
+  /**
+   * Options used to create an auto moderation rule.
+   * @typedef {Object} AutoModerationRuleCreateOptions
+   * @property {string} name The name of the auto moderation rule
+   * @property {AutoModerationRuleEventType} eventType The event type of the auto moderation rule
+   * @property {AutoModerationRuleTriggerType} triggerType The trigger type of the auto moderation rule
+   * @property {AutoModerationTriggerMetadataOptions} [triggerMetadata] The trigger metadata of the auto moderation rule
+   * <info>This property is required if using a `triggerType` of
+   * {@link AutoModerationRuleTriggerTypes.KEYWORD}, {@link AutoModerationRuleTriggerTypes.KEYWORD_PRESET},
+   * or {@link AutoModerationRuleTriggerTypes.MENTION_SPAM}.</info>
+   * @property {AutoModerationActionOptions[]} actions
+   * The actions that will execute when the auto moderation rule is triggered
+   * @property {boolean} [enabled] Whether the auto moderation rule should be enabled
+   * @property {Collection<Snowflake, Role>|RoleResolvable[]} [exemptRoles]
+   * The roles that should not be affected by the auto moderation rule
+   * @property {Collection<Snowflake, GuildChannel|ThreadChannel>|GuildChannelResolvable[]} [exemptChannels]
+   * The channels that should not be affected by the auto moderation rule
+   * @property {string} [reason] The reason for creating the auto moderation rule
+   */
+
+  /**
+   * Creates a new auto moderation rule.
+   * @param {AutoModerationRuleCreateOptions} options Options for creating the auto moderation rule
+   * @returns {Promise<AutoModerationRule>}
+   */
+  async create({
+    name,
+    eventType,
+    triggerType,
+    triggerMetadata,
+    actions,
+    enabled,
+    exemptRoles,
+    exemptChannels,
+    reason,
+  }) {
+    const data = await this.client.api.guilds(this.guild.id)['auto-moderation'].rules.post({
+      data: {
+        name,
+        event_type: typeof eventType === 'number' ? eventType : AutoModerationRuleEventTypes[eventType],
+        trigger_type: typeof triggerType === 'number' ? triggerType : AutoModerationRuleTriggerTypes[triggerType],
+        trigger_metadata: triggerMetadata && {
+          keyword_filter: triggerMetadata.keywordFilter,
+          regex_patterns: triggerMetadata.regexPatterns,
+          presets: triggerMetadata.presets?.map(preset =>
+            typeof preset === 'number' ? preset : AutoModerationRuleKeywordPresetTypes[preset],
+          ),
+          allow_list: triggerMetadata.allowList,
+          mention_total_limit: triggerMetadata.mentionTotalLimit,
+        },
+        actions: actions?.map(action => ({
```

Comment:
```suggestion
        actions: actions.map(action => ({
```
v14: https://github.com/discordjs/discord.js/blob/3160ab0fc75946041cbe7a093032ea10503dfd8c/packages/discord.js/src/managers/AutoModerationRuleManager.js#L128